src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -449,6 +460,164 @@ export default class SourceFileLinter {
+			classInterfaces.initializer && ts.isArrayLiteralExpression(classInterfaces.initializer)) {
+			hasAsyncInterface = classInterfaces.initializer
+				.elements.some((implementedInterface) =>
+					implementedInterface.getText() === "\"sap.ui.core.IAsyncContentCreation\"");
+		}
+
+		let manifestJson: propsRecord = {};
+		if (componentManifest && ts.isPropertyDeclaration(componentManifest) &&
+			componentManifest.initializer && ts.isObjectLiteralExpression(componentManifest.initializer)) {
+			manifestJson = this.extractPropsRecursive(componentManifest.initializer) ?? {};
+		}
+
+		let rootViewAsyncFlag: boolean | undefined;
+		let routingAsyncFlag: boolean | undefined;
+		let rootViewAsyncFlagNode: ts.Node | undefined;
+		let routingAsyncFlagNode: ts.Node | undefined;
+
+		if (manifestJson.manifest?.value === "\"json\"") { // The manifest is an external manifest.json file
+			const parsedManifestContent =
+				JSON.parse(this.#manifestContent ?? "{}") as SAPJSONSchemaForWebApplicationManifestFile;
+
+			const {rootView, routing} = parsedManifestContent["sap.ui5"] ?? {} as JSONSchemaForSAPUI5Namespace;
+			// @ts-expect-error async is part of RootViewDefFlexEnabled and RootViewDef
+			rootViewAsyncFlag = rootView?.async as boolean;
+			routingAsyncFlag = routing?.config?.async;
+		} else {
+			/* eslint-disable @typescript-eslint/no-explicit-any */
+			const instanceOfPropsRecord = (obj: any): obj is propsRecord => {
+				return !!obj && typeof obj === "object";
+			};
+
+			let manifestSapui5Section: propsRecord | undefined;
+			if (instanceOfPropsRecord(manifestJson.manifest?.value) &&
+				instanceOfPropsRecord(manifestJson.manifest.value["\"sap.ui5\""].value)) {
+				manifestSapui5Section = manifestJson.manifest.value["\"sap.ui5\""].value;
+			}
+
+			if (instanceOfPropsRecord(manifestSapui5Section) &&
+				instanceOfPropsRecord(manifestSapui5Section?.rootView?.value) &&
+				typeof manifestSapui5Section?.rootView?.value.async?.value === "boolean") {
+				rootViewAsyncFlag = manifestSapui5Section?.rootView?.value.async?.value;
+				rootViewAsyncFlagNode = manifestSapui5Section?.rootView?.value.async?.node;
+			}
+
+			if (instanceOfPropsRecord(manifestSapui5Section) &&
+				instanceOfPropsRecord(manifestSapui5Section?.routing?.value) &&
+				instanceOfPropsRecord(manifestSapui5Section?.routing?.value.config?.value) &&
+				typeof manifestSapui5Section?.routing?.value.config?.value.async?.value === "boolean") {
+				routingAsyncFlag = manifestSapui5Section?.routing?.value.config?.value.async?.value;
+				routingAsyncFlagNode = manifestSapui5Section?.routing?.value.config?.value.async?.node;
+			}
+		}
+
+		if (!hasAsyncInterface) {
+			if (rootViewAsyncFlag !== true || routingAsyncFlag !== true) {
+				this.#reporter.addMessage({
+					node: classDesc,
+					severity: LintMessageSeverity.Error,
+					ruleId: "ui5-linter-no-sync-loading",
+					message: "Use of sync loading for Component's views",
```

Comment:
It would be very helpful to get a better message in such case. I have tested this on a project that does not use routing (openui5-sample-app) and was confused that there was an error.

Reply:
My point here was not only the false-positive, but also the fact that it mentions "Root View **and Routing** are not configured to load targets asynchronously", even when my `manifest.json` does not use any routing features.

Reply:
This is still open. The message might suggest to a user that they have to specify "sap.ui5/routing/config" in order to solve this issue. But if routing is not used, the routing configuration does not need to be provided at all.

Reply:
Thanks! Fixed it with https://github.com/SAP/ui5-linter/pull/73/commits/cacf9007fbdae7bef2423685eae7c8feeae952e0